controllers/spacebindingcleanup/spacebinding_cleanup_controller_test.go

```diff
@@ -117,25 +117,65 @@ func TestDeleteSpaceBinding(t *testing.T) {
 				require.Error(t, err)
 			})
 
-			t.Run("kubesaw-authenticated SpaceBinding is removed when redhat space is missing", func(t *testing.T) {
-				fakeClient := test.NewFakeClient(t, sbPublicViewerRedhatView, toolchainconfig)
+			t.Run("kubesaw-authenticated SpaceBinding requeues when redhat space is missing and SpaceBinding is too young", func(t *testing.T) {
+				// given
+				sbYoung := sbPublicViewerRedhatView.DeepCopy()
+				sbYoung.CreationTimestamp = metav1.NewTime(time.Now().Add(-5 * time.Second))
+				fakeClient := test.NewFakeClient(t, sbYoung, toolchainconfig)
 				reconciler := prepareReconciler(t, fakeClient)
 
 				// when
-				res, err := reconciler.Reconcile(context.TODO(), requestFor(sbPublicViewerRedhatView))
+				res, err := reconciler.Reconcile(context.TODO(), requestFor(sbYoung))
+
+				// then
+				require.NoError(t, err)
+				require.True(t, res.RequeueAfter > 24*time.Second && res.RequeueAfter <= 25*time.Second) // approximately 25 seconds left
```

Comment:
it's better to use the built-in `Greater` and `Less` functions:
```suggestion
				assert.Greater(t, res.RequeueAfter, 24*time.Second)
				assert.Less(t, res.RequeueAfter, 25*time.Second)
```